demo/site/src/pageTypes/Page.tsx

```diff
@@ -38,35 +38,29 @@ export const pageQuery = gql`
     ${topMenuPageTreeNodeFragment}
 `;
 
-export async function loader({
-    client,
-    pageTreeNodeId,
-    scope,
-}: {
-    client: GraphQLClient;
-    pageTreeNodeId: string;
-    scope: GQLPageTreeNodeScopeInput;
-}): Promise<unknown> {
+export async function loader({ client, pageTreeNodeId, scope }: PageTypeLoaderOptions) {
     return client.request<GQLPageQuery>(pageQuery, {
         pageTreeNodeId,
         domain: scope.domain,
         language: scope.language,
     });
 }
 
-export default function Page(props: GQLPageQuery): JSX.Element {
+export default function Page(props: InferPageTypeLoaderPropsType<typeof loader>): JSX.Element {
```

Comment:
inspired by nextjs's `InferGetStaticPropsType`, this infers the return from the loader without explicit typing anywhere needed